File: ReactJS/FinalProject/server/server/user_app/urls.py
from django.urls import path

from server.user_app.views.authentication import LoginView, LogoutView, CheckAuthenticationView
from server.user_app.views.user_delete import DeleteUserProfilePicture
from server.user_app.views.user_list import UserListView
from server.user_app.views.user_info import UserInfoAPIView
from server.user_app.views.user_create import UserCreateView
from server.user_app.views.user_update import UserUpdateAPIView

urlpatterns = (
    path('all/', UserListView.as_view(), name='users-all'),
    path('create/', UserCreateView.as_view(), name='user-create'),
    path('login/', LoginView.as_view(), name='user-login'),
    path('logout/', LogoutView.as_view(), name='user-logout'),
    path('check-auth/', CheckAuthenticationView.as_view(), name='user-authentication'),
    path('info/', UserInfoAPIView.as_view(), name='user-info'),
    path('edit/', UserUpdateAPIView.as_view(), name='user-edit'),
    path('delete-profile-picture/', DeleteUserProfilePicture.as_view(), name='user-delete-profile-picture'),
    # path('', UserProductsView.as_view(), name='index'),
    # path('token/', include([
    #     path('', AppTokenObtainPairView.as_view(), name='token_obtain_pair'),
    #     path('refresh/', TokenRefreshView.as_view(), name='token_refresh'),
    #     path('blacklist/', TokenBlacklistView.as_view(), name='token_blacklist'),
    # ])),
    # path('user/', include([
    #     path('create/', UserCreateView.as_view(), name='user-create'),
    #     path('<int:pk>/', UserRetrieveUpdateView.as_view(), name='user-retrieve-update'),
    # ])),
    # path('users/', UserListView.as_view(), name='user-list'),
)

File: ReactJS/FinalProject/client/src/utils/Urls.js
const BASE_URL = 'http://localhost:8000/api'

const Urls = {
    baseUrl: BASE_URL,
    CRSFToken: `${BASE_URL}/crsf-token/`,
    index: `${BASE_URL}/`,
    user: {
        details: `${BASE_URL}/user/`,
        register: `${BASE_URL}/user/create/`,
        login: `${BASE_URL}/user/login/`,
        logout: `${BASE_URL}/user/logout/`,
        authentication: `${BASE_URL}/user/check-auth/`,
        profile: `${BASE_URL}/user/info/`,
        update: `${BASE_URL}/user/edit/`,
        deleteProfilePicture: `${BASE_URL}/user/delete-profile-picture/`,
    },


    products: `${BASE_URL}/products/`,
    productCreate: `${BASE_URL}/product/create/`,
    productDetail: `${BASE_URL}/product/:id/`,
    productInfo: `${BASE_URL}/product/:id/info/`,
    productInfoCreate: `${BASE_URL}/product/:id/info/create/`,
    productBarcodes: `${BASE_URL}/product/:id/barcodes/`,
    productBarcodesCreate: `${BASE_URL}/product/:id/barcodes/create/`,
    productUploadImage: `${BASE_URL}/product-info/upload/`,
    groups: `${BASE_URL}/groups/`,
    groupDetail: `${BASE_URL}/group/:id/`,
    groupCreate: `${BASE_URL}/group/create/`,
    groupDelete: `${BASE_URL}/group/:id/`,
}

export default Urls;